src/code-snippet/code-snippet.component.ts

```diff
@@ -137,11 +138,19 @@ export class CodeSnippet {
 	@ViewChild("code") code;
 
 	get shouldShowExpandButton() {
-		return this.code ? this.code.nativeElement.getBoundingClientRect().height > 255 : false;
+		// Checks if `hasExpand` button has been initialized in `AfterViewInit` before detecting whether or not to
+		// show the expand button when the code displayed in the component changes during the life of the component.
+		// This is to avoid the `ExpressionChangedAfterItHasBeenCheckedError`.
+		if (this.hasExpandButton === null) {
+			return this.hasExpandButton;
```

Comment:
does `null` behave differently to `false` in angular templates?

Reply:
That is to prevent the change in `hasExpandButton` before it gets changes in `ngAfterViewInit`. Without the check it gives the expression changed error, and if its checking for false, the expand button will not return if it goes away until the component gets instantiated again

Reply:
Huh, so angular does treat a `null` differently than a `false` then? Interesting!

Reply:
In the template it does treat `null` as `false` and the expand button doesn't show up, when `hasExpandButton` is `null` it means that it has not been initialized in `ngAfterViewInit` yet and it will stay null until it does to avoid the expression changed error

Reply:
Okay, so for clarity - if it was `false`, and stayed as `false` until `hasExpandButton` initialized, would that also work or no?

Reply:
If there was a way for `shouldShowExpandButton` to detect if it has been changed after `ngAfterViewInit` then yes, I currently have it so that it detects if it has not changed if it is still null.

Reply:
okay ... I think I understand, we should be good to go then 👍